csharp/src/Google.Protobuf.Test/MessageParsingHelpers.cs

```diff
@@ -96,5 +102,26 @@ public static void AssertReadingMessage(MessageParser parser, byte[] bytes, Acti
             Assert.AreEqual(message, parsedStream);
             additionalAssert?.Invoke(parsedStream);
         }
+
+        public static void AssertWritingMessage(IMessage message)
+        {
+            // serialize using CodedOutputStream
+            var bytes = message.ToByteArray();
+
+            int messageSize = message.CalculateSize(); 
+            Assert.AreEqual(message.CalculateSize(), bytes.Length);
+
+            // serialize using IBufferWriter and check it leads to the same output
+            var bufferWriter = new ArrayBufferWriter<byte>();
+            message.WriteTo(bufferWriter);
+            Assert.AreEqual(bytes, bufferWriter.WrittenSpan.ToArray());
+
+            // serialize into a single span and check it leads to the same output
+            var singleSpan = new Span<byte>(new byte[messageSize]);
```

Comment:
You could replace this with `Span<byte> singleSpan = stackalloc byte[messageSize];` and test writing to a stack allocated array.

Probably better to test that out by itself. This method is used all over the place and there is a limit on how large you can stackalloc.

Reply:
I made it test serialization to a stackalloc whenever the size is <256 bytes.